pkg/providers/azure.go

```diff
@@ -319,21 +328,52 @@ func (sc *SetUpCmd) createFederatedCredentials() error {
 
 }
 
-func (sc *SetUpCmd) getAppObjectId() error {
+func (wrapper *AzureCredentialWrapper) OAuthToken() string {
+	decoded, err := base64.StdEncoding.DecodeString(wrapper.token)
+	if err != nil {
+		fmt.Println("decode error:", err)
+		return ""
+	}
+	return string(decoded)
+}
+
+func (sc *SetUpCmd) getAppObjectId(ctx context.Context) error {
 	log.Debug("Fetching Azure application object ID")
-	getObjectIdCmd := exec.Command("az", "ad", "app", "show", "--only-show-errors", "--id", sc.appId, "--query", "id")
-	out, err := getObjectIdCmd.CombinedOutput()
+
+	cred, err := cred.GetCred()
+	// Get a token from the credentials
+	token, err := cred.GetToken(ctx, policy.TokenRequestOptions{
+		// okay to hardcode to PublicCloud since we should never deploy to anything else in public OSS repo
```

Comment:
I don't think this should always be true for Draft but I'm fine with it for now.